### DISCUSSION TITLE
Make remote URL dynamic to allow multiple params for API call

### DISCUSSION THREAD

**Discussion by niels-heinemann** [digitalfortress-tech/typeahead-standalone#36] [Ideas]
What a supercool library, thanks a load.
I think it would be beneficious to allow using a function/callback for url property of remote and prefetch datasources. I have to pass dynamically changnd filter values to the reote api. 
If you do not think this would be a valuable new feature, can you provide a precallremote callback?

**Comment by niketpathak** [comment]
Hello @niels-heinemann, Thank you for your comment. Actually there exists a mechanism to pass more data using the `requestOptions` parameter of the remote/prefetch sources. (https://typeahead.digitalfortress.tech/#config?id=source)

An example using a remote source is shown below for further clarity
```js
typeahead({
  input: document.querySelector('.search'),
  source: {
    identifier: 'name',
    remote: {
      url: 'https://restcountries.com/v2/name/%QUERY',
      wildcard: '%QUERY',
      requestOptions: {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ name: 'Clark Kent' }),
      },
    },
  },
  highlight: true,
  limit: 10
})
```
As per the above example, we pass additional data as a key-value pair (`{ name: 'Clark Kent' }`)
You can use the same analogy for `prefetch` option as well.

Hope that helps!

**Comment by niels-heinemann** [comment]
Yeah, thx, I know. This is vanilla fetch options. But I'm afraud I have to use GET parameters. Please do not invest too much time. As I needed this feature I switched to another library today.

**Reply by niketpathak** [reply]
Ah, yes, you are right, if you are to use "GET" params, then it is not possible at the moment. But thank you for highlighting this, we will implement it in a future version.

**Reply by niels-heinemann** [reply]
Would be too nice, thx!

**Comment by niketpathak** [comment]
For future readers who need this feature, it has been implemented in [v4.13.0](https://github.com/digitalfortress-tech/typeahead-standalone/releases/tag/v4.13.0). It allows you to use a callback as a URL in remote/prefetch sources. For usage details, refer [documentation](https://typeahead.digitalfortress.tech/#config?id=source).

**Comment by niels-heinemann** [comment]
Jesses, man! That's quite fast. Thx a load.
